python/interpreter_extensions.bzl

```diff
@@ -0,0 +1,55 @@
+# Copyright 2023 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"Module extensions for use with bzlmod and pip interpreters"
+
+load("@pythons_hub//:interpreters.bzl", "INTERPRETER_LABELS")
+
+def _pypi_ext_impl(mctx):
+    # TODO do we want to run this more than once with multiple
+    # python interpreters?
+    # If so we need to do the for loop
```

Comment:
It's likely we'll want to have the `multi_version` example working under `bzlmod` at some point, so we may want to create multiple repos here.